src/sandstorm/sandstorm-http-bridge.c++

```diff
@@ -1195,7 +1201,59 @@ private:
     auto req = apiCap.restoreRequest();
     req.setToken(percentDecode(buf));
 
-    return req.send().getCap().castAs<Identity>();
+    // Don't directly call `getCap()` on the `RemotePromise` because we want to make sure
```

Comment:
This breaks pipelining in several cases, but it looks like there's only one call site that actually cares about catching exceptions. Can you instead go back to returning a pipelined capability, but call `.whenResolved()` on it in the case where you're actually testing the capability's validity? This should be equivalent.